src/typescript/sdk/src/client/emojicoin-client.ts

```diff
@@ -0,0 +1,332 @@
+ * The `utils` functions provides several commonly used utility functions.
+ *
+ * @example
+ * ```typescript
+ * const emojis: MarketSymbolEmojis = ["🌊"];
+ * const emojicoin = new EmojicoinClient();
+ * const account = Account.generate();
+ * const integrator = account.accountAddress;
+ * await emojicoin.register(account, emojis, { integrator });
+ * const buyArgs = {
+ *  inputAmount: 100n,
+ *  minOutputAmount: 1n,
+ *  integrator,
+ *  integratorFeeRateBPs: 0,
+ * };
+ * await emojicoin.buy(account, emojis, buyArgs);
+ * ```
+ */
+export class EmojicoinClient {
+  public aptos: Aptos;
+
+  public liquidity = {
+    provide: this.provideLiquidity.bind(this),
+    remove: this.removeLiquidity.bind(this),
+  };
+
+  public utils = {
+    emojisToHexStrings: this.emojisToHexStrings.bind(this),
+    emojisToHexSymbol: this.emojisToHexSymbol.bind(this),
+    getEmojicoinInfo: this.getEmojicoinInfo.bind(this),
+    getTransactionEventData: this.getTransactionEventData.bind(this),
+  };
+
+  public rewards = {
+    buy: this.buyWithRewards.bind(this),
+    sell: this.sellWithRewards.bind(this),
+  };
+
+  constructor(args?: { aptos?: Aptos }) {
+    this.aptos = args?.aptos ?? getAptosClient().aptos;
+  }
+
+  async register(
+    registrant: Account,
+    emojis: MarketSymbolEmojis,
+    args: {
+      integrator: AccountAddressInput;
+    },
+    options?: Options
+  ) {
+    const response = await RegisterMarket.submit({
+      aptosConfig: this.aptos.config,
+      registrant,
+      emojis: this.emojisToHexStrings(emojis),
+      integrator: args.integrator,
+      ...options,
+    });
+    const res = this.getTransactionEventData(response);
+    return {
+      ...res,
+      registration: {
+        event: expect(res.events.marketRegistrationEvents.at(0), Expect.Register.Event),
+        model: expect(res.models.marketRegistrationEvents.at(0), Expect.Register.Model),
+      },
+    };
+  }
+
+  async chat(
+    user: Account,
+    emojis: SymbolEmoji[],
+    args: {
+      chatEmojis: AnyEmoji[];
+      emojiIndicesSequence: number[];
+    },
+    options?: Options
+  ) {
+    const chatEmojis = getEmojisInString(args.chatEmojis.join("")) as SymbolEmoji[];
+    const emojiBytes = this.emojisToHexStrings(chatEmojis);
+    const emojiIndicesSequence = new Uint8Array(args.emojiIndicesSequence);
+
+    const response = await Chat.submit({
+      aptosConfig: this.aptos.config,
+      user,
+      emojiBytes,
+      emojiIndicesSequence,
+      ...options,
+      ...this.getEmojicoinInfo(emojis),
+    });
+    const res = this.getTransactionEventData(response);
+    return {
+      ...res,
+      chat: {
+        event: expect(res.events.chatEvents.at(0), Expect.Chat.Event),
+        model: expect(res.models.chatEvents.at(0), Expect.Chat.Model),
+      },
+    };
+  }
+
+  async buy(
+    swapper: Account,
+    emojis: SymbolEmoji[],
+    args: {
+      inputAmount: Uint64;
+      minOutputAmount: Uint64;
+      integrator: AccountAddressInput;
```

Comment:
Same here, how about abstracting this away ? Maybe get the default one from env if none is passed ?

Reply:
I do agree here that the base class (vs the test-only class, as mentioned in the other comment) could pass in an `integrator` and `integratorFeeRateBPs` upon instantiation to avoid having to specify them later.

Possibly some other fields that would mean having to specify fewer args as well